packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -740,6 +752,110 @@ const Conversation = WebexPlugin.extend({
+   * @param {object} query object containing query string values to be appended to the url
+   * @param {string} query.activityType The type of child activities to return.
+   * The value is one of 'reply', 'edit', 'cardAction', 'reaction', 'reactionSummary', 'reactionSelfSummary'
+   * @param {number} [query.limit] The limit of child activities that can be returned per request
+   * @param {boolean} [query.latestActivityFirst] Sort order for the child activities
+   * @param {boolean} [query.includeParentActivity] Enables the parent activity to be returned in the activity list
+   * @param {string} [sinceDate] Get all child activities after this date
+   * @param {string} [maxDate] Get all child activities before this date
+   * @returns {Promise<Array>}
+   */
+  async listChildActivitiesByParentId(conversationUrl, activityParentId, query) {
+    const params = {
+      method: 'GET',
+      url: `${conversationUrl}/parents/${activityParentId}`,
+      qs: query
+    };
+
+    return this.request(params);
+  },
+
+  /**
+   * Returns a list of _all_ child activities for a given parentId within a given conversation
+   * @param {object} options
+   * @param {string} [options.conversationUrl] targeted conversation URL
+   * @param {string} [activityParentId] parent id of edit activities or thread activities
+   * @param {object} [query] object containing query string values to be appended to the url
+   * @param {string} [query.activityType] The type of child activities to return.
+   * The value is one of 'reply', 'edit', 'cardAction', 'reaction', 'reactionSummary', 'reactionSelfSummary'
+   * @param {number} [query.limit] The limit of child activities that can be returned per request
+   * @param {boolean} [query.latestActivityFirst] Sort order for the child activities
+   * @param {boolean} [query.includeParentActivity] Enables the parent activity to be returned in the activity list
+   * @param {string} [sinceDate] Get all child activities after this date
+   * @param {string} [maxDate] Get all child activities before this date
+   * @returns {Promise<Array>}
+   */
+  async listAllChildActivitiesByParentId(options = {}) {
+    const {conversationUrl, activityParentId, query} = options;
+
+    const initialResponse = await this.listChildActivitiesByParentId(conversationUrl, activityParentId, query);
+
+    let page = new Page(initialResponse, this.webex);
+
+    const items = [...page.items];
+
+    while (page.hasNext()) {
+      // eslint-disable-next-line no-await-in-loop
+      page = await page.next();
+      for (const activity of page) {
+        items.push(activity);
+      }
+    }
+
+    if (last(items).published < items[0].published) {
+      items.reverse();
+    }
+
+    return items;
+  },
+
+  /**
+   * @param {string} conversationUrl
+   * @param {string} activityParentId
+   * @param {object} query
+   * @returns {Promise}
+   */
+  async getReactionSummaryByParentId(conversationUrl, activityParentId, query) {
```

Comment:
Not sure I understand the value of this function when `listChildActivitiesByParentId` already can filter activity types?

Reply:
This function calls a different endpoint than listChildActivitiesByParentId. I've just spoken to the engineer who pulled reactions into the web client, and he is vouching for this implementation VS a child activities implementation due to the convo API returning both types of reaction summary here, rather than a server round trip to call each one specifically.